pkg/virt-config/configuration_test.go

```diff
@@ -725,8 +725,8 @@ var _ = Describe("test configuration", func() {
 		Entry("when networkConfiguration set, should equal to result",
 			v1.KubeVirtConfiguration{
 				NetworkConfiguration: &v1.NetworkConfiguration{
-					NetworkInterface:                  string(v1.SlirpInterface),
-					PermitSlirpInterface:              pointer.P(true),
+					NetworkInterface:                  string(v1.DeprecatedSlirpInterface),
+					DeprecatedPermitSlirpInterface:    pointer.P(true),
```

Comment:
Should the slirp be allowed to be default?

Reply:
On upgrade the KV spec does not change.
There is no restriction imposed to set KV in order not to break existing workloads or the upgrade itself.

New VM/VMI will not accept this configuration and the Admin will need to remove Slirp from the KV spec after the upgrade in order to avoid having Slirp.

Reply:
>On upgrade the KV spec does not change.
There is no restriction imposed to set KV in order not to break existing workloads or the upgrade itself.

I don't think we would break compatibility, the config is internal, it would just drop to default. I am not sure how this works with other bindings.... 

>New VM/VMI will not accept this configuration and the Admin will need to remove Slirp from the KV spec after the upgrade in order to avoid having Slirp.

Does this means that new VM/I will be denied or that they will no use the config?

Reply:
> > On upgrade the KV spec does not change.
> > There is no restriction imposed to set KV in order not to break existing workloads or the upgrade itself.
> 
> I don't think we would break compatibility, the config is internal, it would just drop to default. I am not sure how this works with other bindings....

Internal? The Kubevirt CR is not internal.
Anything there will remain as is and it will be reflected when creating new VM/VMI.
If the configuration there is not supported, it will fail validation.

> > New VM/VMI will not accept this configuration and the Admin will need to remove Slirp from the KV spec after the upgrade in order to avoid having Slirp.
> 
> Does this means that new VM/I will be denied or that they will no use the config?

New VM/VMI are denied if the configuration is not supported (i.e. set Slirp interface based on the Kubevirt CR spec). Existing VM/VMI are not affected.
This is no different from defining Slirp explicitly on the VM/VMI.

Reply:
>Internal? The Kubevirt CR is not internal.

Yes but I am commenting about virt-config that is either from configmap or kubevirt cr(the latter lately...). It also has an internal validation on these external sources as these sources can be submitted to the system in invalid form. What I was trying to say in the previous comments is that the validation still allows the slirp. This semantically doesn't make much sense to me. 

>Anything there will remain as is and it will be reflected when creating new VM/VMI.
If the configuration there is not supported, it will fail validation.

This doesn't sounds better than falling to default binding

Reply:
> Yes but I am commenting about virt-config that is either from configmap or kubevirt cr(the latter lately...). It also has an internal validation on these external sources as these sources can be submitted to the system in invalid form. What I was trying to say in the previous comments is that the validation still allows the slirp. This semantically doesn't make much sense to me.

The validation of VM/VMI to reject Slirp is limited to creation and not updates.
The config is used to calculate defaults also for updates, which should continue passing.

Therefore, I do not think it will be correct to validate the config and reject Slirp at this level.
VM/VMI that are running should not get rejected.

If you still think it is possible to both reject Slirp here and keep existing workloads running (and not fail validation on update), we can improve as a follow up.

> > Anything there will remain as is and it will be reflected when creating new VM/VMI.
> > If the configuration there is not supported, it will fail validation.
> 
> This doesn't sounds better than falling to default binding

It is not better, it is a limitation due to the requirement to support existing VMs running even when their manifest is updated (and therefore validated).